### DISCUSSION TITLE
Starting out - Import parts before BOMs?

### DISCUSSION THREAD

**Discussion by 4jag** [inventree/InvenTree#2758] [Q&A] [answered]
When starting out with Inventree, do I need to _first_ populate the parts, _before_ I upload a BOM? 

I have very limited knowledge of the general practice of managing a part DBs and BOMs. I've gone to bat for our division IT to provide a tool while they tie SAP to Aras, so I'm trying to setup Inventree at an existing site.

I ask because field matching seems to be a problem when uploading a BOM. I had it in my head that I'd be importing a BOM, and that would lead to the creation of the parts that appear in the BOM. But now I question if that's the right approach. So I started looking for a place to add Inventree-part fields. We have lots of different price types, overage types, ... (of course these excel BOMs have a LOT of fields. Some will be obsolete once a database is running, for example a field for whether a PO is approved will move to a different table, etc.).  In addition to the typical part parameters we have several specialized fields. So I'm not sure if can add new fields. I'd like to add fields to other tables too, and tho I looked all through the settings and API doc page, it's still not clear if custom fields would be a configuration task, or if that requires developing inside the code base. My guess is that the fields will have to be carefully crafted since there may be sub and sub-sub assemblies.

So I'm looking for some guidance on getting started, and importing some BOM-centric data into Inventree... Whether I have to first populate Manufactures, Suppliers, Parts and other data before I import BOMs, or if I can start with BOM file import. If there's a section in the docs that walks through importing a whole range of related data, then I missed it.

Thank you

**Comment by matmair** [comment, marked as answer]
We have something like that called Part Parameters: https://inventree.readthedocs.io/en/latest/part/parameter/
If you need logic you will need to go into the code / write a plugin for that.

Regarding part importing, I suggest this paragraph: https://inventree.readthedocs.io/en/latest/part/part/#part-import
In general, you add the parts before you insert the BOM.

InvenTree will not be able to replace a massive ERP like SAP or a fully-fledged Enterprise PLM like Aras.

**Reply by 4jag** [reply]
Re parameters -- I saw that. ok. I'll try my best.

Re parts -- So is that's a Yes? Do I need to populate my parts database before I import BOMs? If in BOMs, parts are references to part records in a parts table, then the answer would be yes(?). In order to have a PO with parts that appear on multiple BOMs, then this is a fully relational database across parts, POs, BOMs, suppliers, etc.  ? -- I'm can;'t tell which fields are free-text, if Inventree creates parts records when a BOM is importred (with reasonable field matches), or if Inventree does string matching on reports/search, which are just text values vs record-IDs with vales fetched from another table, and how the BOM import matches these up. I can see that for someone starting from scratch this would all be intuitive...

Re Inventree isn't going to replace Aras... I know, I'm just trying to give our engineers something better than emailing modified xlsx file back and fourth. (We already have many other product lines fully integrated into SAP+PLS, but this group is going to take a while longer)

**Reply by matmair** [reply]
@4jag you do need to create the parts before they can be assigned. That can be by import or manual. InvenTree is a parts database - so yes, it uses a relational database.
